app/routes/user/password.tsx

```diff
@@ -0,0 +1,211 @@
+    if (formRef.current && fetcher.state === 'submitting') {
+      formRef.current.reset()
+    } else if (formRef && formRef.current) {
+      // this works around autofill on the old password field for all browsers
+      setOldPassword((formRef.current[0] as HTMLInputElement).value || '')
+    }
+  }, [fetcher.state])
+
+  return (
+    <>
+      <h1>Reset Password</h1>
+      <>
+        <fetcher.Form method="POST" ref={formRef}>
+          <Label htmlFor="oldPassword">Old Password</Label>
+          <TextInput
+            data-focus
+            className={errorMessage && !oldPassword ? 'usa-input--error' : ''}
+            name="oldPassword"
+            id="oldPassword"
+            type="password"
+            placeholder="Old Password"
+            onChange={(e) => {
+              setOldPassword(e.target.value)
+            }}
+          />
+          {errorMessage === 'NotAuthorizedException' && (
+            <span className="text-red">Invalid Password</span>
+          )}
+          {errorMessage === 'LimitExceededException' && (
+            <span className="text-red">
+              Attempts Exceeded. Please try again later.
+            </span>
+          )}
+          <Label htmlFor="newPassword">New Password</Label>
+          <InputGroup error={isNewPasswordTouched && !valid}>
+            <TextInput
+              name="newPassword"
+              id="newPassword"
+              className={valid ? 'usa-input--success' : ''}
+              type={visible ? 'text' : 'password'}
+              placeholder="New Password"
+              onChange={(e) => {
+                setNewPassword(e.target.value)
+                setIsNewPasswordTouched(true)
+              }}
+              value={newPassword}
+            />
+            <div className="usa-input-suffix" aria-hidden="true">
+              {visible ? (
+                <Icon.VisibilityOff
+                  className="margin-y-2"
+                  aria-label="Hide password"
+                  onClick={() => {
+                    setVisible(!visible)
+                  }}
+                />
+              ) : (
+                <Icon.Visibility
+                  className="margin-y-2"
+                  aria-label="Show password"
+                  onClick={() => {
+                    setVisible(!visible)
+                  }}
+                />
+              )}
+            </div>
+          </InputGroup>
+          <ButtonGroup>
+            <Button
+              disabled={shouldDisableSubmit}
+              type="submit"
+              className="margin-y-2"
+            >
+              Reset Password
+            </Button>
+            {fetcher.state !== 'idle' && (
+              <>
+                <Spinner className="margin-top-3" /> Saving...
+              </>
+            )}
+            {fetcher.state === 'idle' && fetcher.data === null && (
+              <>
+                <Icon.Check className="margin-top-3" color="green" /> Saved
```

Comment:
If you look up on line 158 you'll see that I put a margin on the button. I did this to give it a little space between the last text input and the button. I think that since they are children of the same parent, when the spinner/check render, they render without the same margin as their sibling element. 
<img width="732" alt="Screen Shot 2023-07-18 at 3 41 27 PM" src="https://github.com/nasa-gcn/gcn.nasa.gov/assets/6545874/e044b6c6-7f46-4e02-ba27-d8e4b9e4f4a2">
<img width="748" alt="Screen Shot 2023-07-18 at 3 41 42 PM" src="https://github.com/nasa-gcn/gcn.nasa.gov/assets/6545874/0287c500-4cd9-4132-976c-0e27b76e38b5">